### PR TITLE
Minor fixes in layer_gradient_x_activation and saliency

### PR DESCRIPTION
**Pull request by NarineK**
Minor fixes:
 - It looks like I still had the old naming of `uses_input_marginal_effects` in saliency - fixed that.
- As I was running` LayerGradientActivation` for word indices, I noticed that applying gradients functionality is failing for non-decimal input types. I modified it and added a warning for those cases. Layer methods should still work for those cases and we do not necessarily always need `requires_grad` for the inputs of the layer attribution algorithms.

### CODE REVIEW

**Review by vivekmig — APPROVED** [review]
Looks great, thanks for fixing this 👍 !

File: captum/_utils/gradient.py [hunk 1]
```diff
@@ -32,7 +32,14 @@ def apply_gradient_requirements(inputs: Tuple[Tensor, ...]) -> List[bool]:
     for index, input in enumerate(inputs):
         assert isinstance(input, torch.Tensor), "Given input is not a torch.Tensor"
         grad_required.append(input.requires_grad)
-        if not input.requires_grad:
+        inputs_dtype = input.dtype
+        if inputs_dtype == torch.long or inputs_dtype == torch.int:
```
[the remark this answers is not in the document]
Comment (NarineK) [reply captum/_utils/gradient.py:36]: Yeah, it didn't cover all cases - just the common ones.
It looks like in python API we can check on the floating point and complex types:
https://github.com/pytorch/pytorch/blob/master/torch/tensor.py#L547
This can potentially suffice.
